### PR TITLE
packaging: adjustment for 2.0 branch

### PR DESCRIPTION
**Pull request by bergwolf**
The pr makes necessary adjustment to:
1. let packaging work for the `2.0-dev` branch on release tagging and static tarball building
2. add github action yaml to trigger artifacts build upon 2.0 release tagging

### CODE REVIEW

File: versions.yaml [hunk 1]
```diff
@@ -138,8 +138,8 @@ assets:
     url: "https://github.com/kata-containers/kata-containers/tools/osbuilder"
     architecture:
       aarch64:
-        name: &default-initrd-name "ubuntu"
-        version: &default-initrd-version "18.04"
+        name: &default-initrd-name "fedora"
+        version: &default-initrd-version "30"
```
Comment (fidencio) [inline versions.yaml:142]: If we're going to use Fedora for that, I'd recommend we use the newest supported version (32) or even rawhide for that.

Using Fedora 30 is not exactly the best decision, as it already reached its EOL.

Comment (bergwolf) [reply versions.yaml]: It was just for my local testing. I've fixed up alpine rootfs.

**Comment by bergwolf** [comment]
I've tested this locally up till `kata-deploy` docker uploading stage in my personal forked repository. Cannot really test it in my repository any more. Maybe we can merge it and fix the remaining part of github actions incrementally?

ref: https://github.com/bergwolf/kata-containers/runs/893918239?check_suite_focus=true

**Comment by bergwolf** [comment]
The two failed CI are due to their own setup issues and can be ignored ATM.

File: ci/install_musl.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+set -e
+
+install_aarch64_musl() {
+	local arch=$(uname -m)
+	if [ "${arch}" == "aarch64" ]; then
+		local musl_tar="${arch}-linux-musl-native.tgz"
+		local musl_dir="${arch}-linux-musl-native"
+		poshd /tmp
+		curl -sLO https://musl.cc/${musl_tar}
```
Comment (jodh-intel) [inline ci/install_musl.sh:15]: Nit: Running network operation as `root`? Or is it: I thought we used `sudo` in all these scripts to become `root`?

**Review by jodh-intel — APPROVED** [review]
Thanks @bergwolf. This is a massive PR so it's hard to review everything, hence I think we need more eyes on this.

lgtm

/cc @jcvenegas, @amshinde, @devimc.

File: ci/install_musl.sh (hunk 2 above)
Comment (bergwolf) [reply ci/install_musl.sh:15]: Good point! Thanks!

Comment (bergwolf) [reply ci/install_musl.sh:15]: Turns out we don't have sudo there. Let me retry and see if we are already running root in CI.

**Comment by fidencio** [comment]
In the last Architectures Meeting (Aug 25th 2020), we agreed on not providing the packages from 2.0 onward, in favour of using kata-deploy.

Knowing that, and considering https://github.com/kata-containers/kata-containers/pull/628, we could just drop some of the patches of this PR.

What do you think?

**Comment by bergwolf** [comment]
@fidencio I did add some obvious change to the deb/rpm specs like removing `kata-proxy`/`kata-shim`/`kata-ksm-throttler`. But most of the PR is to make `kata-deploy` work for 2.0. There could be some conflicts between $418 and #628. We can deal with them depending on which PR lands first.

**Comment by fidencio** [comment]
> @fidencio I did add some obvious change to the deb/rpm specs like removing `kata-proxy`/`kata-shim`/`kata-ksm-throttler`. But most of the PR is to make `kata-deploy` work for 2.0. There could be some conflicts between #418 and #628. We can deal with them depending on which PR lands first.

Sure thing. I'm just mentioned #628 in order to save you some time, so you don't end up spending too much time on something that will end up being removed sooner than later.

**Comment by bergwolf** [comment]
@fidencio Yeah, thanks for it. I did not intend to fix deb/rpm building in this PR ;)

**Comment by bergwolf** [comment]
The PR is ready for review. The failed CI `jenkins-ci-ubuntu-1804-cloud-hypervisor-k8s-containerd-2.0 ` is CI setup issue and unrelated to this PR.

**Review by teawater — APPROVED** [review]

**Comment by Jakob-Naucke** [comment]
@bergwolf sorry for the very late necrobump, but I have a question:
In https://github.com/kata-containers/kata-containers/pull/418/commits/76c18aa3454ee7acdc6870b779d617ae26fdced8, you wrote

> Since we always build musl kata-agent, there is no need to build
it inside a musl container. We can just build on the host and then
copy the binary to the target rootfs.
> There are still a lot to clean up and it should be made so for ALL
target distros instead of just alpine. But this is at least working
for alpine first.

Did you mean we shouldn't build the agent inside a container ever again? If so, can we remove all the build-dependency packages from the osbuilder Dockerfiles?
Or can we re-enable building inside a container for other distros?

**Comment by Jakob-Naucke** [comment]
@bergwolf ^^

**Comment by fidencio** [comment]
> @bergwolf sorry for the very late necrobump, but I have a question:
> In [76c18aa](https://github.com/kata-containers/kata-containers/commit/76c18aa3454ee7acdc6870b779d617ae26fdced8), you wrote
> 
> > Since we always build musl kata-agent, there is no need to build
> > it inside a musl container. We can just build on the host and then
> > copy the binary to the target rootfs.
> > There are still a lot to clean up and it should be made so for ALL
> > target distros instead of just alpine. But this is at least working
> > for alpine first.
> 
> Did you mean we shouldn't build the agent inside a container ever again? If so, can we remove all the build-dependency packages from the osbuilder Dockerfiles?
> Or can we re-enable building inside a container for other distros?

Hey, not @bergwolf here, but I'd prefer building everything inside a container and never ever poluting the host OS by installing packages on it.

So, please, I'd strongly back up the idea of re-enabling the agent build inside a container, and then also move the kernel and even runtime builds to be done inside a container.

**Pull request merged**